### PR TITLE
feat(worker): Add Geometry refiner

### PR DESCRIPTION
**Pull request by dongri**
# Overview
Add geometry refiner action

## Summary by CodeRabbit

- **New Features**
    - Introduced a new "Geometry Refiner" entity to process and refine geometric data.
    - Added a workflow configuration for the "geometry_refiner" process to manage and refine geometry-related data through a series of interconnected actions.

### CODE REVIEW

File: worker/crates/action-processor/src/geometry/refiner.rs [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+type RefinedGeometry<T, Z> = (Vec<Point<T, Z>>, Vec<LineString<T, Z>>, Vec<Polygon<T, Z>>);
+
+impl Refiner {
+    fn refine_city_gml(_geos: &CityGmlGeometry) -> Vec<CityGmlGeometry> {
+        // not implemented
+        Vec::new()
+    }
+
+    fn refine_2d(geos: &Geometry2D) -> Vec<Geometry<f64, NoValue>> {
```
Comment (miseyu) [inline worker/crates/action-processor/src/geometry/refiner.rs:149]:
```suggestion
    fn refine_2d(geos: &Geometry2D) -> Vec<Geometry2D<f64>> {
```

File: worker/crates/action-processor/src/geometry/refiner.rs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        for line in lines {
+            geometories.push(Geometry::LineString(line));
+        }
+        for polygon in polygons {
+            geometories.push(Geometry::Polygon(polygon));
+        }
+        geometories
+    }
+
+    fn refine_3d(geos: &Geometry3D) -> Vec<Geometry<f64, f64>> {
```
Comment (miseyu) [inline worker/crates/action-processor/src/geometry/refiner.rs:164]:
```suggestion
    fn refine_3d(geos: &Geometry3D) -> Vec<Geometry3D<f64>> {
```

File: worker/crates/action-processor/src/geometry/refiner.rs [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+impl Refiner {
+    fn refine_city_gml(_geos: &CityGmlGeometry) -> Vec<CityGmlGeometry> {
+        // not implemented
+        Vec::new()
+    }
+
+    fn refine_2d(geos: &Geometry2D) -> Vec<Geometry<f64, NoValue>> {
+        let (points, lines, polygons) = Self::refine_geometry(geos.clone());
+        let mut geometories: Vec<Geometry<f64, NoValue>> = Vec::new();
```
Comment (miseyu) [inline worker/crates/action-processor/src/geometry/refiner.rs:151]:
```suggestion
        let mut geometories: Vec<Geometry2D<f64>> = Vec::new();
```

File: worker/crates/action-processor/src/geometry/refiner.rs [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        for point in points {
+            geometories.push(Geometry::Point(point));
```
Comment (miseyu) [inline worker/crates/action-processor/src/geometry/refiner.rs:153]:
```suggestion
            geometories.push(Geometry2D::Point(point));
```

File: worker/crates/action-processor/src/geometry/refiner.rs [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        }
+        for line in lines {
+            geometories.push(Geometry::LineString(line));
```
Comment (miseyu) [inline worker/crates/action-processor/src/geometry/refiner.rs:156]:
```suggestion
            geometories.push(Geometry2D::LineString(line));
```

File: worker/crates/action-processor/src/geometry/refiner.rs [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        }
+        for polygon in polygons {
+            geometories.push(Geometry::Polygon(polygon));
```
Comment (miseyu) [inline worker/crates/action-processor/src/geometry/refiner.rs:159]:
```suggestion
            geometories.push(Geometry2D::Polygon(polygon));
```

File: worker/crates/action-processor/src/geometry/refiner.rs [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+        }
+        geometories
+    }
+
+    fn refine_3d(geos: &Geometry3D) -> Vec<Geometry<f64, f64>> {
+        let (points, lines, polygons) = Self::refine_geometry(geos.clone());
+        let mut geometories: Vec<Geometry<f64, f64>> = Vec::new();
+        for point in points {
+            geometories.push(Geometry::Point(point));
```
Comment (miseyu) [inline worker/crates/action-processor/src/geometry/refiner.rs:168]:
```suggestion
            geometories.push(Geometry3D::Point(point));
```

File: worker/crates/action-processor/src/geometry/refiner.rs [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+        }
+        for line in lines {
+            geometories.push(Geometry::LineString(line));
```
Comment (miseyu) [inline worker/crates/action-processor/src/geometry/refiner.rs:171]:
```suggestion
            geometories.push(Geometry3D::LineString(line));
```

File: worker/crates/action-processor/src/geometry/refiner.rs [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+        }
+        for polygon in polygons {
+            geometories.push(Geometry::Polygon(polygon));
```
Comment (miseyu) [inline worker/crates/action-processor/src/geometry/refiner.rs:174]:
```suggestion
            geometories.push(Geometry3D::Polygon(polygon));
```

**Comment by miseyu** [comment]
@dongri 
Commented on.
Please check it.

**Comment by dongri** [comment]
@miseyu Fixed
https://github.com/reearth/reearth-flow/pull/350/commits/9c933a85dcb9844d64acf4ce5ef66c2badc50022

**Review by miseyu — APPROVED** [review]
Thank you so much!!
Please merge it.

LGTM

**Pull request merged**